bindgen/clang.rs

```diff
@@ -1268,77 +1278,69 @@ impl Type {
         self.canonical_type() == *self
     }
 
-    #[inline]
-    fn clang_size_of(&self, ctx: &BindgenContext) -> c_longlong {
-        match self.kind() {
-            // Work-around https://bugs.llvm.org/show_bug.cgi?id=40975
-            CXType_RValueReference | CXType_LValueReference => {
-                ctx.target_pointer_size() as c_longlong
-            }
-            // Work-around https://bugs.llvm.org/show_bug.cgi?id=40813
-            CXType_Auto if self.is_non_deductible_auto_type() => -6,
-            _ => unsafe { clang_Type_getSizeOf(self.x) },
-        }
-    }
-
-    #[inline]
-    fn clang_align_of(&self, ctx: &BindgenContext) -> c_longlong {
-        match self.kind() {
-            // Work-around https://bugs.llvm.org/show_bug.cgi?id=40975
-            CXType_RValueReference | CXType_LValueReference => {
-                ctx.target_pointer_size() as c_longlong
-            }
-            // Work-around https://bugs.llvm.org/show_bug.cgi?id=40813
-            CXType_Auto if self.is_non_deductible_auto_type() => -6,
-            _ => unsafe { clang_Type_getAlignOf(self.x) },
-        }
-    }
-
     /// What is the size of this type? Paper over invalid types by returning `0`
     /// for them.
     pub(crate) fn size(&self, ctx: &BindgenContext) -> usize {
-        let val = self.clang_size_of(ctx);
-        if val < 0 {
-            0
-        } else {
-            val as usize
-        }
+        self.fallible_size(ctx).unwrap_or(0)
     }
 
     /// What is the size of this type?
     pub(crate) fn fallible_size(
         &self,
         ctx: &BindgenContext,
     ) -> Result<usize, LayoutError> {
-        let val = self.clang_size_of(ctx);
-        if val < 0 {
-            Err(LayoutError::from(val as i32))
-        } else {
-            Ok(val as usize)
+        match self.kind() {
+            // Work-around for https://github.com/llvm/llvm-project/issues/40320.
+            CXType_RValueReference | CXType_LValueReference => {
+                Ok(ctx.target_pointer_size())
+            }
+            // Work-around for https://github.com/llvm/llvm-project/issues/40159.
+            CXType_Auto if self.is_non_deductible_auto_type() => {
+                Err(LayoutError::Undeduced)
+            }
+            _ => {
+                let size = unsafe { clang_Type_getSizeOf(self.x) };
+                if size < 0 {
+                    Err(LayoutError::from(size))
+                } else {
+                    Ok(size
+                        .try_into()
+                        .expect("type size does not fit in `usize`"))
```

Comment:
I still don't think having `.expect(...)` adds anything more over `.unwap()` since the failure is obvious and the default unwrap error message is good. But no harm in having it.

Reply:
Seems like I missed a few.